### PR TITLE
Update homebrew formula to match homebrew release formula

### PR DESCRIPTION
**Pull request by tzinsky**

### CODE REVIEW

**Review by mppf — APPROVED** [review]

**Comment by mppf** [comment]
@tzinsky also be sure to update your PR message

File: util/packaging/homebrew/chapel-main.rb [hunk 1]
```diff
@@ -31,6 +37,16 @@ class Chapel < Formula
   # LLVM is built with gcc11 and we will fail on linux with gcc version 5.xx
   fails_with gcc: "5"
 
+  # Work around Homebrew 11-arm64 CI issue, which outputs unwanted objc warnings like:
+  # objc[42134]: Class ... is implemented in both ... One of the two will be used. Which one is undefined.
+  # These end up incorrectly failing checkChplInstall test script when it checks for stdout/stderr.
+  # TODO: remove when Homebrew CI no longer outputs these warnings or 11-arm64 is no longer used.
+  patch :DATA
+
+  def llvm
```
[the remark this answers is not in the document]
Comment (tzinsky) [reply util/packaging/homebrew/chapel-main.rb:46]: The patch block is there to workaround a homebrew CI issue. They have a comment above the patch indicating why they are doing this and it includes a TODO to remove this once homebrew resolves their issues...

File: util/packaging/homebrew/chapel-main.rb [hunk 2]
```diff
@@ -1,25 +1,31 @@
 class Chapel < Formula
   desc "Programming language for productive parallel computing at scale"
   homepage "https://chapel-lang.org/"
-  url "https://github.com/chapel-lang/chapel/releases/download/1.26/chapel-1.26.0.tar.gz"
-  sha256 "ba396b581f0a17f8da3f365a3f8b079b8d2e229a393fbd1756966b0019931ece"
+#  url "https://github.com/chapel-lang/chapel/releases/download/1.26/chapel-1.26.0.tar.gz"
+#  sha256 "ba396b581f0a17f8da3f365a3f8b079b8d2e229a393fbd1756966b0019931ece"
+  url "http://web.us.cray.com/~tzinsky/tarball/chapel-1.27.0.prerelease.tar.gz"
+  sha256 "a39198bcd0ac66ee01d35390cf10cbcdaa6ea9da09bc979106fdc25b77c9b2c7"
```
[the remark this answers is not in the document]
Comment (tzinsky) [reply util/packaging/homebrew/chapel-main.rb:7]: Sorry, this was my mistake in trying to get this up after finishing my testing.   I am removing the URL and SHA for my test build.

File: util/packaging/homebrew/chapel-main.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (mppf) [reply util/packaging/homebrew/chapel-main.rb:44]: Right, but I was asking about the `def llvm`.

[the remark this answers is not in the document]
Comment (tzinsky) [reply util/packaging/homebrew/chapel-main.rb:44]: Right - that code is used to enumerate the llvm dependencies which we define earlier in the formula and is intended to replace this block:

```
      if OS.mac?
        if MacOS.version > :catalina
          system "echo CHPL_LLVM_CONFIG=#{HOMEBREW_PREFIX}/opt/llvm@13/bin/llvm-config >> chplconfig"
        else
          system "echo CHPL_LLVM_CONFIG=#{HOMEBREW_PREFIX}/opt/llvm@11/bin/llvm-config >> chplconfig"
        end
      else
        system "echo CHPL_LLVM_CONFIG=#{HOMEBREW_PREFIX}/opt/llvm@13/bin/llvm-config >> chplconfig"
      end
```

With this code: 

```
        CHPL_LLVM_CONFIG=#{llvm.opt_bin}/llvm-config
```

I will look at adding some information to clarify what this code block is used for.

**Pull request merged**